### ISSUE TITLE
Password with special characters not working connection in $config['db_dsnw']  variable

### ISSUE THREAD

**Issue by vbware** [roundcube/roundcubemail#7864]
Easy. When we use passwords with special character y mysql database, connections fails. Its easy makes works change database password, but its an inconvenience.
The vble $config['db_dsnw']  locate inside config file not working correctly with username or password with special characters.
I try to change using html_encode or similar functions but doesnt work.
Finally i must change password and now works, but the problem ist located inside config file, i think.

**Comment by alecpl** [comment]
The special characters in password should be url_encoded. What characters did you use and how the DSN string looked like?

**Comment by vbware** [comment]
For example. I configured my database password with a '%' character.. so, if i use this character, roundcube not works and show "DATABASE CONNECTION ERROR'. I thinks happends with '/' character too.
I fixed the error changing database password, but is very hard, because i am migrating systems, and other apps have this character in all of passwords, so i need to change password database, but i must change every passwords config files.
Thanks!
